services/webhook/tests/test_llm_client.py

```
@@ -148,6 +148,31 @@ def staged_post(*args, **kwargs):
     assert len(out.findings) == 1
 
 
+def test_cave_arms_carry_require_keys_json_schema(monkeypatch) -> None:
+    """#609: both Cave arms must send the require-keys findings json_schema
+    (extra_body replaces the default json_object, which ollama maps to a
+    truncation-prone bare format=json - the #544 estate trap, production
+    edition). The schema mirrors _coerce_finding's required fields."""
+    monkeypatch.setenv("GRUG_REVIEW_DEPTH", "deep")
+    seen = []
+
+    def respond(url, **kwargs):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `respond`

(ANN202)